processor/src/chiplets/aux_trace/mod.rs

```diff
@@ -480,74 +559,129 @@ fn build_bitwise_request<E: FieldElement<BaseField = Felt>>(
     is_xor: Felt,
     alphas: &[E],
     row: RowIndex,
+    _debugger: &mut BusDebugger<E>,
 ) -> E {
-    let op_label = get_op_label(ONE, ZERO, is_xor, ZERO);
-    let a = main_trace.stack_element(1, row);
-    let b = main_trace.stack_element(0, row);
-    let z = main_trace.stack_element(0, row + 1);
+    let bitwise_request_message = BitwiseMessage {
+        op_label: get_op_label(ONE, ZERO, is_xor, ZERO),
+        a: main_trace.stack_element(1, row),
+        b: main_trace.stack_element(0, row),
+        z: main_trace.stack_element(0, row + 1),
+    };
```

Comment:
I wonder if it would make sense to move some of this logic into specific message structs similar to how we've moved computation of the value logic there as well. For example, we could have something like:

```rust
impl BitwiseMessage {
    pub fn new_request(is_xor: bool, trace: &MainTrace, row: RowIndex) -> Self {
        ...
    }

    pub fn new_response(trace: &MainTrace, row: RowIndex) -> Self {
        ...
    }
}
```
This way, all message-building logic would be nicely encapsulated in one place.

Reply:
This can be done for the `BitwiseMessage`, but not for all message (e.g. the memory requests don't get their `addr` from the same place, amongst other things). So some messages would be built "directly in the bus" (as in the current implementation), and others would be encapsulated in the message - which I consider a code smell.

Also the message construction logic is very succinct, so I like seeing how the messages are built directly in the bus.

Reply:
As discussed offline, I didn't add constructors, but I split up that file into multiple ones (and it does clear things up a lot).